### PR TITLE
feat: anonymised telemetry to track usage patterns

### PR DESCRIPTION
**Pull request by sidroopdaska**

### CODE REVIEW

File: fam/llm/fast_inference.py [hunk 1]
```diff
@@ -156,8 +163,28 @@ def synthesise(self, text: str, spk_ref_path: str, top_p=0.95, guidance_scale=3.
...
         duration_s = librosa.get_duration(y=audio, sr=sr)
+        real_time_factor = time_to_synth_s / duration_s
         print(f"\nTotal time to synth (s): {time_to_synth_s}")
-        print(f"Real-time factor: {time_to_synth_s / duration_s:.2f}")
+        print(f"Real-time factor: {real_time_factor:.2f}")
+
+        posthog.capture(
+            TelemetryEvent(
+                name="user_ran_tts",
+                properties={
```
Comment (vatsalaggarwal) [inline fam/llm/fast_inference.py:173]: nit: can you add gpu type so we can track against real_time_factor?

File: fam/llm/finetune.py [hunk 2]
```diff
@@ -352,7 +366,14 @@ def get_lr(it):
 
             # termination conditions
             if iter_num > max_iters:
-                break
+                # log end of finetuning event
+                posthog.capture(
+                    TelemetryEvent(
+                        name="user_completed_finetuning",
+                        properties={"finetune_jobid": finetune_jobid},
```
Comment (vatsalaggarwal) [inline fam/llm/finetune.py:373]: add loss?

File: fam/telemetry/__init__.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                with open(self.USER_ID_PATH, "w") as f:
+                    new_user_id = str(uuid.uuid4())
+                    f.write(new_user_id)
+                self._curr_user_id = new_user_id
+            else:
+                with open(self.USER_ID_PATH, "r") as f:
+                    self._curr_user_id = f.read()
+        except Exception:
+            self._curr_user_id = self.UNKNOWN_USER_ID
+        return self._curr_user_id
```
Comment (vatsalaggarwal) [inline fam/telemetry/__init__.py:43]: nit: EOL

File: fam/telemetry/posthog.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+load_dotenv()
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO, handlers=[logging.StreamHandler(sys.stdout), logging.StreamHandler(sys.stderr)])
+
+
+class PosthogClient(TelemetryClient):
+    def __init__(self):
+        self._posthog = Posthog(
+            project_api_key="phc_tk7IUlV7Q7lEa9LNbXxyC1sMWlCqiW6DkHyhJrbWMCS", host="https://eu.posthog.com"
```
Comment (vatsalaggarwal) [inline fam/telemetry/posthog.py:18]: are these safe?

File: pyproject.toml [hunk 5]
```diff
@@ -24,10 +24,13 @@ gradio = "^4.20.1"
 huggingface_hub = "^0.21.4"
 click = "^8.1.7"
 wandb = { version = "^0.16.4", optional = true }
+posthog = "^3.5.0"
+python-dotenv = "^1.0.1"
 
 [tool.poetry.dev-dependencies]
 pre-commit = "^3.7.0"
 pytest = "^8.0.2"
+ipdb = "^0.13.13"
```
Comment (vatsalaggarwal) [inline pyproject.toml:33]: remove? where is this being used here?

File: pyproject.toml [hunk 6]
```diff
@@ -24,10 +24,13 @@ gradio = "^4.20.1"
 huggingface_hub = "^0.21.4"
 click = "^8.1.7"
 wandb = { version = "^0.16.4", optional = true }
+posthog = "^3.5.0"
+python-dotenv = "^1.0.1"
```
Comment (vatsalaggarwal) [inline pyproject.toml:28]: remove? where is this being used here?

**Review by vatsalaggarwal — COMMENTED** [review]
LGTM!

why are new formatting changes getting introduced? I think yours look right, but not sure how the current code got committed?

File: pyproject.toml (hunk 6 above)
Comment (sidroopdaska) [reply pyproject.toml:28]: why remove? these deps are needed

File: pyproject.toml (hunk 5 above)
Comment (sidroopdaska) [reply pyproject.toml:33]: same as below. these deps are required for dev work.

File: fam/telemetry/posthog.py (hunk 4 above)
Comment (sidroopdaska) [reply fam/telemetry/posthog.py:18]: yes, it's safe to use.

**Review by lucapericlp — DISMISSED** [review]

File: pyproject.toml (hunk 6 above)
Comment (vatsalaggarwal) [reply pyproject.toml:28]: i didn't see a `.env` file so was unsure

**Review by vatsalaggarwal — COMMENTED** [review]
lgtm

**Pull request merged**